hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OzoneFileSystem.java

```diff
@@ -59,7 +59,13 @@ public URI getKeyProviderUri() throws IOException {
   @Override
   public DelegationTokenIssuer[] getAdditionalTokenIssuers()
       throws IOException {
-    KeyProvider keyProvider = getKeyProvider();
+    KeyProvider keyProvider;
+    try {
+      keyProvider = getKeyProvider();
+    } catch (IOException ioe) {
+      LOG.error("Error retrieving KeyProvider.", ioe);
+      return null;
+    }
     if (keyProvider instanceof DelegationTokenIssuer) {
```

Comment:
Why don't we just check keyProvider!=null on Line 69 before check if keyProvider instanceof? This way, we don't need other changes. 

Note getKeyProvider() will be eventually fixed when OM returns its KMS uri to the client as part of it service discovery.